javascript/selenium-atoms/text.js

```diff
@@ -56,15 +56,39 @@ core.text.getTextContent_ = function(element, preformatted) {
     }
     return text.replace(/&nbsp/, ' ');
   }
-  if (element.nodeType == goog.dom.NodeType.ELEMENT &&
-      element.nodeName != 'SCRIPT') {
+  if (SHADOW_DOM_ENABLED &&
+      element.nodeType == goog.dom.NodeType.ELEMENT &&
+      element.shadowRoot !== null) {
+    return core.text.getTextContent_(element.shadowRoot, preformatted);
+  }
+  if ((element.nodeType == goog.dom.NodeType.ELEMENT ||
+       element.nodeType == goog.dom.NodeType.DOCUMENT_FRAGMENT) &&
+      element.nodeName != 'SCRIPT' &&
+      element.nodeName != 'STYLE') {
```

Comment:
some web components have style tags in their shadow trees. Polymer is a good example of this.

So, from what I remember, we would end up with CSS in the text content result.